pints/_log_pdfs.py

```diff
@@ -115,6 +115,161 @@ def sample(self, n=1):
         raise NotImplementedError
 
 
+class PooledLogPDF(LogPDF):
+    r"""
+    Combines :math:`m` :class:`LogPDFs<pints.LogPDF>`, each with :math:`n`
+    parameters, into a single LogPDF where :math:`k` parameters are "pooled"
+    (i.e. have the same value for each LogPDF), so that the resulting combined
+    LogPDF has :math:`m (n - k) + k` parameters.
+
+    This is useful for e.g. modelling the time-series of multiple individuals
+    (each individual defines a separate :class:`LogPDF`), and some parameters
+    are expected to be the same across individuals (for example, the noise
+    parameter across different individuals within the same experiment).
+
+    For two :class:`LogPDFs<pints.LogPDF>` :math:`L _1`  and
+    :math:`L _2` with parameters :math:`(\psi _1, \theta _1`) and
+    :math:`(\psi _2, \theta _2`) respectively, a pooling of the
+    :math:`\theta _i` results in a pooled log-pdfs of the form
+
+    .. math::
+        L(\psi _1, \psi _2, \theta | D_1, D_2) =
+            L(\psi _1, \theta | D_1) + L(\psi _2, \theta | D_2),
+
+    where :math:`\theta := \theta _1 = \theta _2`, and :math:`D_i` is the
+    measured time-series of individual :math:`i`.
```

Comment:
Maybe add something like "In this case, we have m=2, k is the size of \theta, and n is the size of psi + k"? The notation of n and k does not directly indicate the notation of the parameters \psi and \theta here.